tests/Unit/CacheInvalidatorTest.php

```diff
@@ -184,8 +184,29 @@ public function testMethodException()
      */
     public function testProxyClientExceptionsAreLogged()
     {
-        $unreachableException = ProxyUnreachableException::proxyUnreachable('http://127.0.0.1', 'Couldn\'t connect to host');
-        $responseException    = ProxyResponseException::proxyResponse('http://127.0.0.1', 403, 'Forbidden');
+        $failedRequest = \Mockery::mock('\Psr\Http\Message\RequestInterface')
+            ->shouldReceive('getHeaderLine')->with('Host')->andReturn('127.0.0.1')
+            ->getMock();
+        $adapterException = \Mockery::mock(
+            '\Http\Adapter\Exception\HttpAdapterException',
+            ['Couldn\'t connect to host']
+        )
+            ->shouldReceive('getRequest')->andReturn($failedRequest)
+            ->getMock();
+        ;
+        $unreachableException = ProxyUnreachableException::proxyUnreachable($adapterException);
+        
+        $response = \Mockery::mock('\Psr\Http\Message\ResponseInterface')
+            ->shouldReceive('getStatusCode')->andReturn(403)
+            ->shouldReceive('getReasonPhrase')->andReturn('Forbidden')
+            ->getMock();
+        $adapterException = \Mockery::mock('\Http\Adapter\Exception\HttpAdapterException')
+            ->shouldReceive('getMessage')->andReturn('Forbidden')
+            ->shouldReceive('getRequest')->andReturn($failedRequest)
+            ->shouldReceive('getResponse')->andReturn($response)
+            ->getMock()
+        ;
+        $responseException = ProxyResponseException::proxyResponse($adapterException);
```

Comment:
PHP-VCR was just an example for fixture generation. This tool should specifically be PSR-7 dependent. From some point of view this can be low level as well.

The testing library sounds great, but I think this fixture library could be used elsewhere as well. For example  @jeromegamez has an awesome Tape Recorder library. Maybe it could use this tool as well.